proposals/0072-feature-gate-threshold-automation.md

```diff
@@ -138,10 +149,16 @@ This is done by using the Get Epoch Stake Syscall.
 
 Nodes should send a transaction containing this instruction at some arbitrary
 point during the epoch at least 128 slots before the end of the epoch and on
-startup after any reboot.
+startup after any reboot. Transactions sent too late (< 128 slots before epoch
```

Comment:
I think this < should be a > ?

Reply:
Nope I think it's correct!

Reply:
Two things:
- Epochs can be shorter than 128 slots, I think the one after genesis starts with 32 slots or so.
- This is (128 * 0.4s = 51.2s) less than a minute. Our cache preparation phase is currently double that, and might expand further. I would make it a lot longer like 30 minutes (30*60s/0.4s = 4500 Slots).

Reply:
> * Epochs can be shorter than 128 slots, I think the one after genesis starts with 32 slots or so.

I think since this process is for new feature activations, it's safe to omit the first epoch after genesis in this design, right?

> * This is (128 * 0.4s = 51.2s) less than a minute. Our cache preparation phase is currently double that, and might expand further. I would make it a lot longer like 30 minutes (30*60s/0.4s = 4500 Slots).

Ok, I think a 4500 slot grace period is acceptable. I think that still leaves plenty of time for all nodes to submit their signals. It's worth mentioning that this would be a program design spec, which can be tweaked in the future.